docs/pipeline_configuration.md

```diff
@@ -0,0 +1,75 @@
+## Overview
+Creating pipelines in Conduit is a simple process with many steps to define the resources and their configurations, this
+could be done through API or Conduit's UI. But to make this process even simpler and to make it possible to load 
+pre-defined pipelines, we introduce **_Pipeline Configurations_**.
+
+Using **_Pipeline Configurations_** you will define all the pipelines you want to be provisioned into Conduit at startup,
+only by creating `.yml` files that define all the pipelines, with their connectors, processors, and configurations.
+
+## Getting started
+create a folder called _"pipelines"_ at the same level as your Conduit binary file, add all your `.yml` files there, 
+then run Conduit using the cmd command: 
+```
+./conduit
+```
+If you have your YAML files at a different directory, simply run conduit with the cli flag `pipelines.path` and point to 
+your directory, ex: 
+```
+./conduit -pipeline.path ../my-directory
+```
+
+_**Note**_: provisioned pipelines are **immutable**, any updates needed on a provisioned pipeline have to be done through
+the configuration file it was provisioned from. you can only control stopping and starting a pipeline through the UI and API.
+
+### YAML file format
+The file in general has two root keys, the "version", and the "pipelines" map. The map consists of other elements like 
+"status" and "name", which are configurations for the pipeline itself.
+
+To create connectors in that pipeline, simply add another map under the pipeline map, and call it "connectors".
+
+To create processors, either add a "processors" map under a pipeline ID, or under a connector ID, depending on its parent.
+check this YAML file example with explanation for each field:
+
+``` yaml
+version: 1.0                    # parser version, the only supported version for now is 1.0 [mandatory]
+pipelines:                      # a map of pipelines IDs and their configurations.
+  pipeline1:                    # pipeline ID, has to be unique.
+    status: running             # pipelines status at startup, either running or stopped. [mandatory]
+    name: pipeline1             # pipeline name. [optional]
```

Comment:
We can mention that Conduit will use the pipeline ID as the name if no name is specified.